openpype/hosts/maya/plugins/publish/extract_fbx_animation.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+import os
+
+from maya import cmds  # noqa
+import pyblish.api
+
+from openpype.pipeline import publish
+from openpype.hosts.maya.api import fbx
+from openpype.hosts.maya.api.lib import namespaced
+
+
+class ExtractFBXAnimation(publish.Extractor):
+    """Extract Rig in FBX format from Maya.
+
+    This extracts the rig in fbx with the constraints
+    and referenced asset content included.
+    This also optionally extract animated rig in fbx with
+    geometries included.
+
+    """
+    order = pyblish.api.ExtractorOrder
+    label = "Extract Animation (FBX)"
+    hosts = ["maya"]
+    families = ["animation.fbx"]
+
+    def process(self, instance):
+        # Define output path
+        staging_dir = self.staging_dir(instance)
+        filename = "{0}.fbx".format(instance.name)
+        path = os.path.join(staging_dir, filename)
+        path = path.replace("\\", "/")
+
+        fbx_exporter = fbx.FBXExtractor(log=self.log)
+        out_set = instance.data.get("animated_skeleton", [])
+        # Export
+        instance.data["constraints"] = True
+        instance.data["skeletonDefinitions"] = True
+        instance.data["referencedAssetsContent"] = True
+
+        fbx_exporter.set_options_from_instance(instance)
+
+        out_set_name = next(out for out in out_set)
+        # temporarily disable namespace
+        namespace = out_set_name.split(":")[0]
+        new_out_set = out_set_name.replace(
+            f"{namespace}:", "")
+        cmds.namespace(relativeNames=True)
+        with namespaced(":" + namespace, new=False) as namespace:
+            path = path.replace("\\", "/")
+            fbx_exporter.export(new_out_set, path)
+        original_relative_names = cmds.namespace(
+            query=True, relativeNames=True)
+        if original_relative_names:
+            cmds.namespace(relativeNames=original_relative_names)
```

Comment:
Please avoid the use of `str.replace()` calls like these. Namespaces can be nested. `:namespace:namespace:namespace:` is technically fine in Maya - what we want to do is just strip of the first that we're entering.

Also, `new_out_set` sounds like you're generating a new node which confuses me a bit.

Additionally you're now querying the `relativeNames` after you've set it so of course it's your value ;) - you want to do that before you're setting it. You're also only setting it IF it's true, but that's not what you want - you want to always revert it to the original value even if it's False.

It should likely be something like this:

```suggestion
        namespace, relative_out_set = out_set_name.split(":", 1)
        original_relative_names = cmds.namespace(
            query=True, relativeNames=True)
        cmds.namespace(relativeNames=True)
        with namespaced(":" + namespace, new=False) as namespace:
            path = path.replace("\\", "/")
            fbx_exporter.export(relative_out_set, path)
        
        cmds.namespace(relativeNames=original_relative_names)
```

This might be made even easier if we add the `relative_names` flag to the `namespaced` context manager so that this code can become:
```python
with namespaced(
    ":" + namespace,
    relative_names=True,
    new=False
) as namespace:
    fbx_exporter.export(relative_out_set, path)
```

Likely even better to maintain like that.

The context manager then would just get an extra argument `relative_names` which defaults to `None` (does nothing), when `False` it forces it to False when `True` it forces it to True and voila. Better code.